src/client/particles.cpp

```diff
@@ -706,13 +760,13 @@ void ParticleManager::clearAll()
 	m_dying_particle_spawners.clear();
 
 	// clear particles
-	for (std::unique_ptr<Particle> &p : m_particles) {
-		// remove scene node
-		p->remove();
-		// delete
-		p.reset();
-	}
 	m_particles.clear();
+
+	// clear buffers
+	for (auto &it : m_particle_buffers) {
+		it->remove();
+		it.reset();
+	}
```

Comment:
Why not
```suggestion
	m_particle_buffers.clear();
```
and use the `ParticleBuffer` dtor to remove itself from the parent? IMO that would be a less error-prone approach.

Reply:
I agree but doesn't work.
For the destructor to run the SceneManager needs to drop its reference to a given buffer, which only happens if `remove` is called.

Reply:
Someone could invent a `irr_scene_node_ptr<>` that calls `remove` before `drop`...